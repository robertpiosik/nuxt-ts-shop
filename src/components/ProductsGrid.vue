<template lang="pug">
	div(:class="$style.grid")
		slot
</template>

<script lang="ts">
import { Vue } from 'vue-property-decorator'

export default class extends Vue {}
</script>

<style lang="scss" module>
.grid {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-gap: 1.5rem;
  @include atMedium {
    grid-template-columns: repeat(3, 1fr);
  }
  @include atLarge {
    grid-template-columns: repeat(4, 1fr);
  }
}
</style>
