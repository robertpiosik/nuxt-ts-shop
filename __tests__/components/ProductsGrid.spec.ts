import { shallowMount } from '@vue/test-utils'
import ProductsGrid from './../../src/components/ProductsGrid.vue'

const slotContent = 'slot content'

const wrapper = shallowMount(ProductsGrid, {
	slots: { default: slotContent }
})

describe('Component: ProductsGrid', () => {
  describe('GENERAL', () => {
    it('should match a snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
    it('should be a Vue instance', () => {
      expect(wrapper.isVueInstance()).toBeTruthy()
    })
  })
  describe('SLOTS', () => {
    it('should render given slot value and nothing else', () => {
      expect(wrapper.text()).toBe(slotContent)
    })
  })
})
