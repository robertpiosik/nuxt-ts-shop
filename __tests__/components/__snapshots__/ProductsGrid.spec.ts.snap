// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsGrid GENERAL should match a snapshot 1`] = `<div class="grid">slot content</div>`;
